jetbrains-core/src/software/aws/toolkits/jetbrains/settings/UpdateLambdaSettings.kt

```diff
@@ -0,0 +1,43 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.settings
+
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+
+@State(name = "updateLambdaSettings", storages = [Storage("aws.xml")])
+class UpdateLambdaSettings : PersistentStateComponent<UpdateSettings> {
+    private var state = UpdateSettings()
+
+    override fun getState(): UpdateSettings? = state
+    override fun loadState(state: UpdateSettings) {
+        this.state = state
+    }
+
+    fun useContainer(stackArn: String): Boolean? = state.updateConfigs[stackArn]?.useContainer
+    fun setUseContainer(stackArn: String, value: Boolean) {
+        state.updateConfigs.computeIfAbsent(stackArn) { UpdateConfig() }.useContainer = value
+    }
+
+    fun bucketName(stackArn: String): String? = state.updateConfigs[stackArn]?.bucketName
```

Comment:
It feel strange to me that you have to pass the `arn` to each of these. To me it makes sense to decouple the persistence from this a bit - so that `getInstance` takes an `arn` and returns a `UpdateConfig` that has getters/setters that talk to the underlying storage. 

So at the call-site you can do something like: 

```kt
val config = UpdateLambdaSettings.getInstance(lambdaArn)
// read values
view.bucket = config.bucketName
// write values
config.bucketName = view.bucket
```

Reply:
I like this, we should use this style in the other places we have maps like this, changed.